File: app/src/main/java/com/jounik_projects/mydatingprofilehelper/data/network/model/GoogleLoginRequest.kt
package com.jounik_projects.mydatingprofilehelper.data.network.model

import com.google.gson.annotations.SerializedName

/**
 * Модель данных для отправки Google токена на бэкенд для авторизации.
 */
data class GoogleLoginRequest(
    @SerializedName("googleToken") // Аннотация для маппинга имени поля в JSON
    val googleToken: String // Токен Google, полученный после успешного входа
)

File: backend/src/MyDatingProfileHelper.Backend.Api/DTOs/GoogleLoginRequest.cs
csharp
// DTO для запроса Google авторизации
namespace MyDatingProfileHelper.Backend.DTOs
{
    /// <summary>
    /// Представляет данные запроса для входа через Google.
    /// </summary>
    public class GoogleLoginRequest
    {
        /// <summary>
        /// Google токен, полученный от клиента Android.
        /// </summary>
        public string? GoogleToken { get; set; }
    }
}